### PR TITLE
Update SPV_AMDX_shader_enqueue to v2

### PR DESCRIPTION
**Pull request by Tobski**
SPIR-V counterpart to https://github.com/KhronosGroup/Vulkan-Docs/pull/2442

### CODE REVIEW

File: extensions/AMD/SPV_AMDX_shader_enqueue.asciidoc [hunk 1]
```diff
@@ -217,47 +270,106 @@ Modify Section 3.21, "BuiltIn", adding the following row to the table:
...
+ +
+_Payload Type_ is the type of each payload in the array. +
+ +
+ See <<OpNodePayloadArrayLengthAMDX,*OpNodePayloadArrayLengthAMDX*>> for getting the length of an array of this type. +
+ +
+A payload array can be allocated by either *OpAllocateNodePayloadsAMDX* to be enqueued as an output, or via *OpVariable* in the *NodePayloadAMDX* storage class to be consumed as an input. +
+ +
+Can be dereferenced using an access chain in the same way as *OpTypeRuntimeArray* or *OpTypeArray*.
+1+|<<Capability,Capability>>: +
+*Shader*
```
Comment (Naghasan) [inline extensions/AMD/SPV_AMDX_shader_enqueue.asciidoc:338]: https://github.com/KhronosGroup/SPIRV-Headers/pull/452 ties it to `ShaderEnqueueAMDX` not `Shader`

File: extensions/AMD/SPV_AMDX_shader_enqueue.asciidoc [hunk 2]
```diff
@@ -268,39 +380,85 @@ _Node Index_
...
+ +
+Query the length of a payload array. Must only be used with input payload arrays or allocated output payload arrays. +
+ +
+_Result_ will be equal to the _Payload Count_ value used to allocate _Payload Array_, or to the number of received payloads if the shader is using *CoalescingAMDX* execution mode. Otherwise, _Result_ will be 1. +
+ +
+_Result Type_ must be an *OpTypeInt* with 32-bit _Width_ and 0 _Signedness_. +
+ +
+_Payload Array_ is a pointer to a payload array previously allocated by *OpAllocateNodePayloadsAMDX*, or declared via *OpVariable* in the *NodePayloadAMDX* storage class as an input.
+1+|<<Capability,Capability>>: +
+*Shader*
```
Comment (Naghasan) [inline extensions/AMD/SPV_AMDX_shader_enqueue.asciidoc:409]: https://github.com/KhronosGroup/SPIRV-Headers/pull/452 ties it to `ShaderEnqueueAMDX` not `Shader`

File: extensions/AMD/SPV_AMDX_shader_enqueue.asciidoc (hunk 1 above)
Comment (yavn) [reply extensions/AMD/SPV_AMDX_shader_enqueue.asciidoc:338]: Yes, thank you, both (this and the other comment) should be `ShaderEnqueueAMDX` as these are the new `*AMDX` instructions.

**Review by Naghasan — APPROVED** [review]

**Comment by bashbaug** [comment]
Merging as discussed in the October 9th teleconference.

**Pull request merged**